kernel/os/src/os.c

```diff
@@ -82,6 +82,14 @@ int g_os_started;
 #define MIN_IDLE_TICKS  (MYNEWT_VAL(OS_IDLE_TICKLESS_MS_MIN) * OS_TICKS_PER_SEC / 1000)
 #define MAX_IDLE_TICKS  (MYNEWT_VAL(OS_IDLE_TICKLESS_MS_MAX) * OS_TICKS_PER_SEC / 1000)
 
+extern int main(int, char **);
+
+__attribute__((weak)) int
+mynewt_main(int argc, char **argv)
```

Comment:
With this **weak** `mynewt_main()` version even though application has own version of mynwet_main(), linker for arm-nono-gcc and xc32-gcc often (not always) chooses this weak version and then complains that `main` is missing.
```
repos/apache-mynewt-core/kernel/os/src/os.c:90: undefined reference to `main'
```
It seems to work better with applications from apache-mynewt-core but when iptest was build for black_vet6 with new mynewt_main() in main.c present weak version was used.
For local applications like blinky or others it newer worked for me.

So I think that if we are going to go this way it would be better to drop this weak version of find some way that is more stable.

Reply:
Solution could be separate package with just one file with weak implementation.